tests/unit/billing/test_enum_definitions.py

```diff
@@ -1,35 +1,99 @@
-from shared.billing import BillingPlan, is_enterprise_cloud_plan
-
-
-def test_billing_enums():
-    assert BillingPlan.users_monthly.db_name == "users-inappm"
-    assert BillingPlan.users_yearly.db_name == "users-inappy"
-    assert BillingPlan.users_free.db_name == "users-free"
-    assert BillingPlan.users_basic.db_name == "users-basic"
-    assert BillingPlan.pr_monthly.db_name == "users-pr-inappm"
-    assert BillingPlan.pr_yearly.db_name == "users-pr-inappy"
-    assert BillingPlan.enterprise_cloud_yearly.db_name == "users-enterprisey"
-    assert BillingPlan.enterprise_cloud_monthly.db_name == "users-enterprisem"
-
-
-def test_get_from_string():
-    assert BillingPlan.from_str("users-inappm") == BillingPlan.users_monthly
-    assert BillingPlan.from_str("users-inappy") == BillingPlan.users_yearly
-    assert BillingPlan.from_str("users-free") == BillingPlan.users_free
-    assert BillingPlan.from_str("users-basic") == BillingPlan.users_basic
-    assert BillingPlan.from_str("users-pr-inappm") == BillingPlan.pr_monthly
-    assert BillingPlan.from_str("users-pr-inappy") == BillingPlan.pr_yearly
-    assert (
-        BillingPlan.from_str("users-enterprisey") == BillingPlan.enterprise_cloud_yearly
-    )
-    assert (
-        BillingPlan.from_str("users-enterprisem")
-        == BillingPlan.enterprise_cloud_monthly
-    )
-
-
-def test_is_enterprise_cloud_plan():
-    assert not is_enterprise_cloud_plan(BillingPlan.pr_monthly)
-    assert not is_enterprise_cloud_plan(BillingPlan.pr_yearly)
-    assert is_enterprise_cloud_plan(BillingPlan.enterprise_cloud_yearly)
-    assert is_enterprise_cloud_plan(BillingPlan.enterprise_cloud_monthly)
+import pytest
+from django.test import override_settings
+
+from shared.billing import BillingPlan, is_enterprise_cloud_plan, is_pr_billing_plan
+from shared.django_apps.codecov_auth.tests.factories import OwnerFactory
+
+
+class TestBillingServiceTestCase(object):
+    @override_settings(IS_ENTERPRISE=True)
+    @pytest.fixture
+    def dbsession(self, db):
+        return db
```

Comment:
I would prefer these to remain fn-based tests rather than class-based tests in general.
Also this specific fixture seems to only be responsible for renaming the magic fn argument?

I believe it would be clearer to just use the existing fixture, or to add the `dbsession` fixture globally so all tests can use it. I would also clean up the used fixtures a bit, though it might not be quite clear which ones are used, as pytest fixtures are often just triggering sideeffects :-(

Reply:
I removed the parts that required the tests to be class-based when I cleaned up the API tests, so function-based tests now make more sense